### ISSUE TITLE
Supabase does not work with react native Google signin

### ISSUE THREAD

**Issue by Eirmas** [supabase/supabase#15946]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug
Supabase describes in their docs for google signin on native:
```
Please take a look at these open-source projects which may help you obtain an ID token directly from the OS:
   [react-native-google-signin/google-signin](https://github.com/react-native-google-signin/google-signin).
```

However this integration is not currently possible. `react-native-google-signin/google-signin` does not provide the raw nonce value before that Supabase expects present. As of right now I cannot find any way of implementing google login with react native and supabase.

See linked issue [here](https://github.com/react-native-google-signin/google-signin/issues/1176)

## To Reproduce

1. Create a Supabase project with Google signin provider
2. Create a React Native project
3. Install `react-native-google-signin/google-signin` and `supabase`
4. Try to integrate them.

**Comment by timdobranski** [comment]
I can confirm; getting error 'AuthApiError: Passed nonce and nonce in id_token should either both exist or not.' when using signInWithIdToken()

**Comment by nick-barth** [comment]
Yes, this is a pretty major blocker for us as well, has anyone found a workaround?  I'm using this [google auth capacitor ](https://github.com/CodetrixStudio/CapacitorGoogleAuth)library, and we also do not have access to the raw nonce.

**Comment by Ngineer101** [comment]
I'm experiencing the same issue. I followed all the steps described in this post: https://supabase.com/blog/native-mobile-auth

Using:
- For Google sign-in: https://github.com/react-native-google-signin/google-signin
- For Apple sign-in: https://github.com/invertase/react-native-apple-authentication

For both auth providers I get the same error: "Passed nonce and nonce in id_token should either both exist or not."

Neither Apple nor Google sign-in works. Any help would be much appreciated.

**Comment by nick-barth** [comment]
> I'm experiencing the same issue. I followed all the steps described in this post: https://supabase.com/blog/native-mobile-auth
> 
> Using:
> 
> * For Google sign-in: https://github.com/react-native-google-signin/google-signin
> * For Apple sign-in: https://github.com/invertase/react-native-apple-authentication
> 
> For both auth providers I get the same error: "Passed nonce and nonce in id_token should either both exist or not."
> 
> Neither Apple nor Google sign-in works. Any help would be much appreciated.

@Ngineer101 it appears that your library allows you to create the raw nonce, so after you've received a idToken you should be able to pass Supabase the raw nonce and it should match.

**Comment by kangmingtay** [comment]
Hey everyone, I chatted with @nick-barth through our support channel and was wondering about the following:

> Seems like the issue here is that after completing the authentication with google, google returns an id token that contains a nonce in it. Based on what I've gathered from this thread, it seems like this nonce is being set by google and you don't have access to the raw value? 
> Is it possible for you to send us an example of the id token issued by google too? Also, it would be great if you can point us to the relevant google auth API endpoint that capacitor uses under the hood and we can investigate further.
